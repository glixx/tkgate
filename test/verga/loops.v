module top #(.xyz(9));
  integer i;
  reg [7:0] a,b,c;
  reg [63:0] s;
  reg q;

  initial
    begin
      a = xyz;
      b = 5;
      c = a + b + 11;
      s = "abcd" + 8 ;

      $display("This is a hello world test string <%s> %d.","string tester",23);
      if (a > b)
        $display("if statment got a 'then' because %d > %d .",a,b);

      i = 0;
      while (i < 5)
        begin
          $display("while i = %d",i);
	  i = i + 1;
        end

      for (i = 3;i < 10;i = i + 1)
	begin
          $display("for i = %d",i);
        end

      $display("top done @%d.",$time);
    end 

  foo #(.delay(1)) f1();
  foo #(.delay(2)) f2();

endmodule

module foo #(.delay(1)) ();
  integer i;

  initial
    begin
      $display("starting %m with delay %d.",delay);
      for (i = 0;i <= 5;i = i + 1)
        #delay $display("in %m i = %d @ %d",i,$time);
    end


endmodule

module bar();
  reg [7:0] a,b,c; 
  parameter y = 8'h6;
  parameter q = {4'h5,4'h9,x,4'h3};
  parameter r = {6{1'b1}};
  parameter h = 13*3;
endmodule
