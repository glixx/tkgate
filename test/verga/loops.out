starting top.f1 with delay 1.
starting top.f2 with delay 2.
This is a hello world test string <string tester> 23.
if statment got a 'then' because 9 > 5 .
while i = 0
while i = 1
while i = 2
while i = 3
while i = 4
for i = 3
for i = 4
for i = 5
for i = 6
for i = 7
for i = 8
for i = 9
top done @0.
in top.f1 i = 0 @ 1
in top.f2 i = 0 @ 2
in top.f1 i = 1 @ 2
in top.f1 i = 2 @ 3
in top.f2 i = 1 @ 4
in top.f1 i = 3 @ 4
in top.f1 i = 4 @ 5
in top.f2 i = 2 @ 6
in top.f1 i = 5 @ 6
in top.f2 i = 3 @ 8
in top.f2 i = 4 @ 10
in top.f2 i = 5 @ 12
